src/throttle/amoc_throttle.erl

```diff
@@ -3,57 +3,63 @@
 -module(amoc_throttle).
 
 %% API
--export([start/2,
-         start/3,
-         start/4,
-         send/3,
-         send/2,
-         send_and_wait/2,
-         wait/1,
-         run/2,
-         pause/1,
-         resume/1,
-         change_rate/3,
-         change_rate_gradually/6,
-         stop/1]).
-
--deprecated([{send_and_wait, 2, "use wait/1 instead"}]).
-
--define(DEFAULT_NO_PROCESSES, 10).
--define(DEFAULT_INTERVAL, 60000). %% one minute
--define(NONNEG_INT(N), (is_integer(N) andalso N >= 0)).
--define(POS_INT(N), (is_integer(N) andalso N > 0)).
+-export([start/2, stop/1,
+         send/2, send/3, wait/1,
+         run/2, pause/1, resume/1, unlock/1,
+         change_rate/2, change_rate_gradually/2]).
 
 -type name() :: atom().
--type rate() :: pos_integer().
+%% Atom representing the name of the throttle.
```

Comment:
it's so confusing to have comments below the code. please move comments above and add blank lines between definitions to make it readable. also, let's make it in edoc format.

Reply:
Nope, not possible to move the comments above and still generate the right documentation. It'd need to be migrated entirely to the newer documentation format.

Reply:
thanks, it's much better with blank lines